### PR TITLE
Java clients v2

### PR DESCRIPTION
**Pull request by oliverhowell**
1st pass at merging content from different PRs into a clean PR - still needs further review and work but important notes:

- see // comments inline for questions and TBC
- WIP restructure of Java client page
- Contains latest cluster routing content TBC re: standalone
- Variables move to antora file
- WIP switch to Google Style Guide

Preview link: https://deploy-preview-1289--hardcore-allen-f5257d.netlify.app/hazelcast/5.5/clients/java

### CODE REVIEW

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:99]: In general, this is true, but do we want to mention possible problems that can happen ? For example, missing APIs due to missing features or missing APIs due to throwing UnsupportedOperationException (and removed therefore) will cause compile problems.

Also, the declerative config and system properties will not work.

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:86]: yes, the only difference can be the necessity of adding the jfrog private repository to pom.xml if EE client is used.

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:241]: as far as the Java Standalone client has routing modes, maybe it's because this documentation is for 5.5.0-beta version?

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:1002]: sorry, I didnt get what this sentence means by "parent configuration".

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:1481]: Why do we use the word "step" here? Are we talking about some internal Hazelcast member process steps?  Or are we trying to talk about the interface methods? If it's about methods, we should call them methods, and  describe as 

"the first method is called when the SocketInterceptor is being initailized" 
"the second method is called when a socket is connected"

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:1494]: we can say, 

"above is an example how to create a SocketInterceptor and add it to client configuration"

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:1556]: the ssl/tls config should be the same 🤔

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:1577]: I don't think cloud is implemented

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:12]:
```suggestion
Hazelcast provides a native {java-client} which you can use to connect to a Hazelcast cluster. `hazelcast-<VERSION>.jar` is bundled in the Hazelcast standard package, so just add `hazelcast-<VERSION>.jar` to your classpath and you can start using this client as if you are using the Hazelcast API.
```

Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:12]:
```suggestion
Hazelcast provides a platform {java-client} which you can use to connect to a Hazelcast cluster. `hazelcast.jar` is bundled in the Hazelcast standard package, so just add `hazelcast.jar` to your classpath and you can start using this client as if you are using the Hazelcast API.
```
I am not sure if native is the terminology we want to use for existing client. We called it as Platform client. We also have native(HD) memory feature which may cause native client to be misunderstood.

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:35]:
```suggestion
NOTE: If you have a Hazelcast {enterprise-product-name} license, you don't need to set the license key in your Hazelcast Java clients to use the xref:getting-started:editions.adoc#features-in-hazelcast-enterprise[{enterprise-product-name} features]. You only have to set it on the member side, and include the `hazelcast-enterprise-<VERSION>.jar` dependency in your classpath.
```

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:81]: full-version should be 5.5.0-BETA

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:99]:
```suggestion
To migrate an application using the {java-client} to the {java-client-new}, you only have to update the dependency as described above.
```

Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:99]: We can have this migration information emphasized, either an info box or bold letters. Or we can have a subsection.

Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:99]: They're all covered right below in https://deploy-preview-1289--hardcore-allen-f5257d.netlify.app/hazelcast/5.5/clients/java#java-client-standalone-beta section

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:86]: Not sure if this is documented - can you use **Add a suggestion** to suggest what/where the draft change would be?

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:241]: Yeah, I saw the conversation here https://github.com/hazelcast/hz-docs/pull/1240#discussion_r1741811784 so put this in ready to be uncommented once confirmed:

@alexey-agapov-hazelcast please confirm

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:1481]: I agree this could be improved - its referring to void init and void onConnect right?

```suggestion
The first method initializes the `SocketInterceptor` using the defined properties.
The second method informs when the socket is connected using the `onConnect` method.
```

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:1494]: Thanks, have added it before the code sample (best practice)

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:1556]: I highlighted this one as it looks like @sergey-serdyukov PR has the changes compared to our original page - please confirm!

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:1577]: Thanks for confirming - standard disclaimer added

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:12]: Thanks, added

Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:12]: Thanks, removed (I thought it might be controversial / inappropriate and it's fine without it)

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:81]: Sounds like we need another variable to track the standalone-full-version if it will diverge from embedded client...updated antora.yml which contains the variables

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:99]: I can add an H4 so we get heading and link too

**Comment by oliverhowell** [comment]
Minor changes from review pushed up, will continue working thru the page as there was some later sections I didn't get to edit yet

File: docs/modules/clients/pages/java.adoc
Comment (alexey-agapov-hazelcast) [reply docs/modules/clients/pages/java.adoc:242]: @oliverhowell the Java Standalone client does support all three routing modes in the current version

File: docs/modules/clients/pages/java.adoc
Comment (alexey-agapov-hazelcast) [reply docs/modules/clients/pages/java.adoc:1559]: Confirmed. ssl/tls settings are identical in both clients

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:128]:
```suggestion
For further information, see <<configure-java-client, Configure the client >>.
```

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:130]:
```suggestion
After completing the client configuration, you must create an `HazelcastClient` instance which will initialize and connect tot he client based on the provided configuration:
```

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:136]:
```suggestion
You can create a distributed map and populate it with some data as follows:
```

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:163]:
```suggestion
IMPORTANT: Currently, the {java-client-new} only supports the IMap distributed data structure.
```

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:418]:
```suggestion
=== Configuring CP direct-to-leader operation routing for clients
```

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:1559]: I think the issue is that in the original PR the Cloud section is removed but the TLS code samples from this section have been kept and moved under TLS section: https://github.com/hazelcast/hz-docs/pull/1240/files#r1761451399

Maybe the code sample works for both cases, will need some guidance here but I could link from here to the programmatic sample for the standalone client?

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [inline docs/modules/clients/pages/java.adoc:88]:
```suggestion
If you are using `hazelcast-enterprise-java-client`, you also need to add this your pom.xml file:

[source,xml,subs="attributes+"]
----
<repositories>
    <repository>
        <id>private-repository</id>
        <name>Hazelcast Private Repository</name>
        <url>https://repository.hazelcast.com/release/</url>
        <releases>
            <enabled>true</enabled>
        </releases>
        <snapshots>
            <enabled>false</enabled>
        </snapshots>
    </repository>
</repositories>
----
```

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [reply docs/modules/clients/pages/java.adoc:86]: something like this https://github.com/hazelcast/hz-docs/pull/1289/files#r1761806719

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [reply docs/modules/clients/pages/java.adoc:1003]: I got it, thanks. For programmatic configuration, I would not use "attribute". Maybe we should use all the main setter methods? I'm not sure.

File: docs/modules/clients/pages/java.adoc
Comment (srknzl) [reply docs/modules/clients/pages/java.adoc:1482]: yes I think so, the change looks fine to me .

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:242]: I'll uncomment the section but see Serkan's question about unisocket mode.

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:97]: Added, thanks

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:1005]: I like the second suggestion, thanks - will update

File: docs/modules/clients/pages/java.adoc
Comment (ihsandemir) [inline docs/modules/clients/pages/java.adoc:2]: Should we also have the link to java client javadoc at here as well.

`* https://docs.hazelcast.org/hazelcast-java-client/{page-latest-supported-java-client-new}/javadoc[Hazelcast {java-client-new} API documentation]`

Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:2]: This is a page variable that triggers the dropdown below the main heading, the page H1 - AFAIK we can't have two links here. i.e. this is a limitation of the existing one page structure. We can have references within the content though (see links further down), which I think is ok for Beta.

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc]: Implemented above suggestion - reopen comment if you have alt suggestion

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc]: This text is out now anyway so resolving

**Comment by oliverhowell** [comment]
Latest changes pushed up - please review @alexey-agapov-hazelcast 

You can also see some comments in source where quick check would be appreciated:
`// check standalone`

File: docs/modules/clients/pages/java.adoc
Comment (alexey-agapov-hazelcast) [inline docs/modules/clients/pages/java.adoc:412]: ClassLoader is supported in {java-client-new}

File: docs/modules/clients/pages/java.adoc
Comment (alexey-agapov-hazelcast) [inline docs/modules/clients/pages/java.adoc:422]: Distributed Executor Service is supported in {java-client-new}

File: docs/modules/clients/pages/java.adoc
Comment (alexey-agapov-hazelcast) [inline docs/modules/clients/pages/java.adoc:1077]: Declarative configuration is supported in {java-client-new}

File: docs/modules/clients/pages/java.adoc
Comment (alexey-agapov-hazelcast) [inline docs/modules/clients/pages/java.adoc:1879]: The following properties are supported by {java-client-new}:

- hazelcast.client.shuffle.member.list
- hazelcast.client.heartbeat.timeout
- hazelcast.client.event.queue.capacity
- hazelcast.client.event.thread.count
- hazelcast.client.heartbeat.interval
- hazelcast.client.invocation.timeout.seconds
- hazelcast.client.invocation.retry.pause.millis
- hazelcast.client.max.concurrent.invocations
- hazelcast.client.invocation.backoff.timeout.millis
- hazelcast.client.io.input.thread.count
- hazelcast.client.io.output.thread.count
- hazelcast.client.io.balancer.interval.seconds
- hazelcast.client.io.write.through
- hazelcast.client.concurrent.window.ms
- hazelcast.client.response.thread.count
- hazelcast.client.response.thread.dynamic
- hazelcast.client.operation.backup.timeout.millis
- hazelcast.client.operation.fail.on.indeterminate.state
- hazelcast.client.statistics.period.seconds
- hazelcast.partitioning.strategy.class
- hazelcast.discovery.enabled
- hazelcast.discovery.public.ip.enabled
- hazelcast.client.connectivity.logging.delay.seconds
- hazelcast.metrics.enabled
- hazelcast.client.metrics.debug.enabled
- hazelcast.client.metrics.jmx.enabled
- hazelcast.client.metrics.collection.frequency

Comment (oliverhowell) [reply docs/modules/clients/pages/java.adoc:1879]: Will add disclaimers for properties not supported

Comment (alexey-agapov-hazelcast) [reply docs/modules/clients/pages/java.adoc:1872]: Maybe we should also change the text in the beginning of the section saying that it's only applicable to the {java-client}

> == Client system properties
> 
> NOTE: This section is only applicable to the {java-client}.
> 
> There are some advanced client configuration properties that help you tune the {java-client}.

**Comment by oliverhowell** [comment]
Can I get an approval on this PR or final comments before I merge?

@alexey-agapov-hazelcast @ihsandemir @srknzl @kwart

**Review by kwart — APPROVED** [review]
The security parts looks good.

**Review by srknzl — APPROVED** [review]

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [inline docs/modules/clients/pages/java.adoc:79]: @ihsandemir you want to change this line?

Comment (ihsandemir) [reply docs/modules/clients/pages/java.adoc]: done

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [inline docs/modules/clients/pages/java.adoc:78]:
```suggestion
To get started using the {java-client-new}, you need to add the `hazelcast-java-client` dependency to your pom.xml, as shown below. You can then start using this client as if
```

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [inline docs/modules/clients/pages/java.adoc:89]:
```suggestion
If you are using `hazelcast-enterprise-java-client`, you need to add the `hazelcast-enterprise-java-client` dependency and private hazelcast repository to your pom.xml file, as shown below:
```

File: docs/modules/clients/pages/java.adoc
Comment (oliverhowell) [inline docs/modules/clients/pages/java.adoc:144]:
```suggestion
For further information on client configuration, see <<configure-java-client, Configure the client>>.
```

**Review by oliverhowell — COMMENTED** [review]
@ihsandemir 2 minor changes and also I spotted a rogue bracket

**Pull request merged**
